src/audit/cache_poisoning.rs

```diff
@@ -0,0 +1,292 @@
+use crate::audit::{audit_meta, WorkflowAudit};
+use crate::finding::{Confidence, Finding, Severity};
+use crate::models::Step;
+use crate::state::AuditState;
+use github_actions_models::common::expr::ExplicitExpr;
+use github_actions_models::common::{Env, EnvValue};
+use github_actions_models::workflow::event::{BareEvent, OptionalBody};
+use github_actions_models::workflow::job::StepBody;
+use github_actions_models::workflow::Trigger;
+use std::ops::Deref;
+use std::sync::LazyLock;
+
+/// The possible toggle types for cache-aware Actions.
+/// A boolean toggle means only one possible literal value (true)
+/// to opt into the cache behavior, whereas a multi-value toggle means
+/// two or more values defining opt in for caching
+#[derive(PartialEq)]
+enum CacheControlToggle {
+    Boolean,
+    MultiValue,
+}
+
+/// The general schema for a cache-aware actions
+struct CacheAwareAction {
+    /// The owner/repo part within the Action full coordinate
+    reference: &'static str,
+    /// The name of the input the toggle caching behavior
+    cache_control_input_name: &'static str,
+    /// Whether this input defines opt-in behavior
+    cache_control_input_defines_opt_in: bool,
+    /// The type of this toggle (Boolean, MultiValue)
+    cache_control_input_toggle: CacheControlToggle,
+    /// Whether this Action adopts caching as the default behavior
+    caching_by_default: bool,
+}
+
+/// The list of know cache-aware actions
+/// In the future we can easily retrieve this list from the static API,
+/// since it should be easily serializable
+static KNOWN_CACHE_AWARE_ACTIONS: LazyLock<Vec<CacheAwareAction>> = LazyLock::new(|| {
+    vec![
```

Comment:
It might be worth adding `PyO3/maturin-action` to this list as well, since `sccache: true` enables caching via `sccache` + GHA :slightly_smiling_face: 

(Our own `pypi.yml` uses this, which is not ideal :sweat_smile:)

Reply:
Added [here](https://github.com/woodruffw/zizmor/pull/294/commits/accbdbdb447e0788b180e85f1deb3b5ff4a286c7)